src/main/java/seedu/cakecollate/model/orderitem/OrderItem.java

```diff
@@ -0,0 +1,81 @@
+package seedu.cakecollate.model.orderitem;
+
+import static seedu.cakecollate.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents an OrderItem.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class OrderItem {
+
+    private final Type type;
+    private final Cost cost;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public OrderItem(Type type, Cost cost) {
+        requireAllNonNull(type, cost);
+        this.type = type;
+        this.cost = cost;
+    }
+
+    public Type getType() {
+        return type;
+    }
+
+    public Cost getCost() {
+        return cost;
+    }
+
+    /**
+     * Returns true if both orders have the same type of cake.
+     * This defines a weaker notion of equality between two orders.
+     */
+    public boolean isSameOrderItem(OrderItem otherOrder) {
```

Comment:
Do you need both isSameOrderItem and equals? You could consider using the equals method only because the same order will probably have the same cost also.

Reply:
I am not really sure whether there are any use cases for equals. Probably will leave it in for now in case it is needed and remove it later if it is not used.